### PR TITLE
Re-adding User Deletion

### PR DESCRIPTION
**Pull request by NomadNaomie**
Admins can now navigate to user pages and access the User Delete tab to delete users. They will be required to reauthenticate 

Delete Tab
image

Reauthentication Modal
image

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -117,8 +117,12 @@ def delete_project_proposal
   end
 
   def delete_user
-    User.find(params[:id]).destroy
-    redirect_to root_path, notice: 'User Deleted!'
+    if User.authenticate(params[:username_email], params[:password]).admin?
```
Comment (art29) [inline app/controllers/admin/users_controller.rb:120]: I would change this line to `if @user.admin? && @user.pword == password`. The reason being that this way the email doesn't need to be re-inputed.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -510,4 +524,27 @@
...
+        <button type="button" class="btn-close" data-bs-dismiss="modal"></button>
+      </div>
+      <div class="modal-body">
+          <div class="mb-3 attach">
+          <%= form_tag(delete_user_admin_users_path(id: @repo_user.id), method: :delete)  do %>
+            <span> Please re-enter your password to confirm your identity.</span>
+            <%= hidden_field_tag "id", @repo_user.id%>
+            <%= label_tag :delete_user, "Are you sure you want to delete this user?", class:"form-label mb-3" %>
+            <%= password_field_tag :password, nil, placeholder: t('sessions.login.password'), class: 'form-control mb-3' %>
+            <%= text_field_tag :username_email, params[:username_email], placeholder: t('sessions.login.username_or_email'), class: 'form-control mb-3', autofocus: true %>
```
Comment (art29) [inline app/views/users/show.html.erb:543]: I would remove the email

**Review by art29 — APPROVED** [review]
Looks good, maybe just remove the email field for ease of use

**Comment by NomadNaomie** [comment]
There are a lot of implications we've discussed, pending further discussions before we can move ahead with this.

**Review by art29 — APPROVED** [review]
Great Work on this! It works really well!

**Pull request merged**